packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_model.dart

```diff
@@ -163,13 +168,120 @@ class IosDomainError extends DomainError {
+    'Applinks details paths format',
+    'This test checks that the `applinks.details.paths` property is an array of strings.',
+  );
+
+  static const detailsDefaultsFormat = AASAfileFormatSubCheck(
+    'Applinks details default format',
+    'This test checks that the `applinks.details.defaults` property only holds '
+        'the keys caseSensitive and percentEncoded. Ref - '
+        'https://developer.apple.com/documentation/bundleresources/applinks/details/default',
+  );
+
+  static const detailsDefaultsPercentEncodedFormat = AASAfileFormatSubCheck(
+    'Applinks defaults percent encoded format',
+    'This test checks that the `applinks.details.defaults.percentEncoded` property only holds a boolean.',
+  );
+
+  static const detailsDefaultsCaseSensitiveFormat = AASAfileFormatSubCheck(
+    'Applinks defaults case sensitive format',
+    'This test checks that the `applinks.details.defaults.caseSensitive` property only holds a boolean.',
+  );
+
+  static const componentFormat = AASAfileFormatSubCheck(
+    'Applinks details components format',
+    'This test checks that the `applinks.details.components` property is formatted properly. Ref - '
+        'https://developer.apple.com/documentation/bundleresources/applinks/details/components',
+  );
+
+  static const componentPathFormat = AASAfileFormatSubCheck(
+    'Applinks details components path format',
+    'This test checks that the `applinks.details.components.path` property only holds a string.',
+  );
+
+  static const componentQueryFormat = AASAfileFormatSubCheck(
+    'Applinks details components query format',
+    'This test checks that the `applinks.details.components.query` property only holds either a string or object of type `applinks.Details.Components.Query`.',
+  );
+
+  static const componentFragmentFormat = AASAfileFormatSubCheck(
+    'Applinks details components fragment format',
+    'This test checks that the `applinks.details.components.fragment` property only holds a string.',
+  );
+
+  static const componentExcludeFormat = AASAfileFormatSubCheck(
+    'Applinks details components exclude format',
+    'This test checks that the `applinks.details.components.exclude` property only holds a boolean.',
+  );
+  static const componentPercentEncodedFormat = AASAfileFormatSubCheck(
+    'Applinks details components percent encoded format',
+    'This test checks that the `applinks.details.components.percentEncoded` property only holds a boolean.',
+  );
+  static const componentCaseSensitiveFormat = AASAfileFormatSubCheck(
+    'Applinks details components case sensitive format',
+    'This test checks that the `applinks.details.components.caseSensitive` property only holds a boolean.',
+  );
+  static const componentCommentFormat = AASAfileFormatSubCheck(
+    'Applinks details components comment format',
+    'This test checks that the `applinks.details.components.comment` property only holds a string.',
```

Comment:
nit we have a lot of duplicated strings here. We could reduce the duplication by adding a helper method:
```
String propertyTypeMessage({required String property, required String expectedType}) {
  return 'This test checks that the `$property` property only holds a $expectedType.';
}
```